src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+        if ( header.getSortOrder() != SAMFileHeader.SortOrder.coordinate )
+            throw new GATKException("The BAM must be coordinate sorted.");
+
+        // Process the input data to produce a map of Kmer -> List of breakpoint IDs.
+        // (A breakpoint ID is just an arbitrary identifier for the putative breakpoint.)
+        // See the doc for findClusters to see a description of how this big Spark pipeline works.
+        final Broadcast<Map<Kmer,List<Long>>> broadcastMap = ctx.broadcast(findClusters(ctx));
+
+        // Process the input again, this time pulling all reads that contain kmers associated with a given breakpoint,
+        // and writing those reads into a separate FASTQ for each breakpoint.
+        getGoodReads()
+            .mapPartitionsToPair(readItr -> new ReadPuller(readItr,broadcastMap.value()))
+            .groupByKey()
+            .foreach(tupl -> writeFastq(tupl._1,tupl._2.iterator(),outputDir));
+    }
+
+    public JavaRDD<GATKRead> getGoodReads() {
+        return getUnfilteredReads().filter(read ->
+            !read.isSecondaryAlignment() && !read.isSupplementaryAlignment() &&
+                    !read.isDuplicate() && !read.failsVendorQualityCheck());
+    }
+
+    // Read a file of kmers to ignore.
+    // Each line must be exactly KLEN characters long, and must match [ACGT]*.
+    private Set<Kmer> readKillSet()
+    {
+        final Set<Kmer> killSet = new HashSet<>();
+
+        try
+        {
+            final BufferedReader rdr = new BufferedReader(new FileReader(kmerKillSet));
+            String line;
+            while ( (line = rdr.readLine()) != null )
+            {
+                if ( line.length() != KLEN )
+                    throw new GATKException("Kmer kill set contains a line of length " + line.length() +
+                            " but we were expecting K=" + KLEN);
+
+                final StringKmerizer kmerizer = new StringKmerizer(line, KLEN);
+                if ( !kmerizer.hasNext() )
+                    throw new GATKException("Unable to kmerize the kmer kill set string '" + line + "'.");
+
+                killSet.add(kmerizer.next());
+            }
+        }
+        catch ( IOException e )
+        {
+            throw new GATKException("Unable to read kmer kill set.", e);
+        }
+
+        return killSet;
+    }
+
+    // Produce a map of Kmer -> List of unique breakpoint IDs.
+    // This is done by looking for reads that appear to be split, or for reads that are part of discordant pairs.
+    // When a sufficient number of these reads are mapped onto reference near one another, those reads are taken
+    // together as evidence of a putative breakpoint.
+    // Each such breakpoint is numbered, and the reads that supported that breakpoint are kmerized.
+    // Kmers that appear on a kill-list are ignored.
+    // This produces a map of Breakpoint ID -> List of Kmers, which is inverted to produce the final output.
```

Comment:
in general use javadoc for all method and class comments, you don't need to use complete javadoc, it's better to leave out useless `@param` tags and the like.  Private code won't be exposed through the doc, but it will make it easier for developers since the IDE will pick up the descriptions for private methods.  It also lets you use `{@link } tags which lets you directly reference variables in your javadoc and results in errors if they get renamed which lets you keep doc in sync much more easily.